### DISCUSSION TITLE
EntityRuler additional properties

### DISCUSSION THREAD

**Discussion by lfoppiano** [explosion/spaCy#9445] [Help: Best practices]
Hi all, 
  I'm using the entity ruler for a quick matching system. In particular, I'm interested in adding additional information in each match. I guess the patterns have a fixed set of properties (id, pattern, label). Is there a way to carry out more information? 

E.g. my matches are as follow: 
``
{"label":"space-groups","pattern":"P6_3mc","type":"hexagonal"}
``
and I would like to find a way to get the "type" somewhere in the entities that are extracted. Is there a way?

Thank you in advance

**Comment by polm** [comment]
One of the main use cases of the ID is to refer to an external source of data. So you can use the ID `space-hex` and then you have some data:

```
space_db = {"space-hex": {"type":"hexagonal"}, ...}
```

And when you get a match you use the ID to look up the data in your data store, which can be a dict, or a DB, or anything. You can't attach more attributes directly to the match results, but the intent is that the ID takes care of any use case like that.

Does that make sense?

**Comment by lfoppiano** [comment]
Dear @polm, indeed it makes sense. Thanks for the tip! :-)
